### ISSUE TITLE
Impressive port!

### ISSUE THREAD

**Issue by ghost** [richardassar/SampleRNN_torch#2]
Impressive port, I'm new to either of these frameworks (theano or Torch) but I can tell significant work went into the port, thanks for sharing it!

Curious, what was your motivation? For example, do you expect better perf, or maybe easier multi-GPU support, etc?

Ive thought of porting to tensorflow, but low prio for me, I guess it would be a good learning project.

**Comment by richardassar** [comment]
Thanks a lot for your comments, I spent a lot of time and effort on this so
it's great to hear someone acknowledge my work.

To answer your question, my motivations were numerous.

Working on this helped me understand Theano a lot better because I use
Torch at work mainly.

It also helped to really understand the SampleRNN model and the finer
details of how to train it.

Training times seem faster for me (but I need to measure this to be
certain). Sampling times are definitely faster though.

I'm writing code that will visualise the hidden activations during
generation to create some interesting videos, this seemed easier to do with
an implementation written by myself. I'll push this once it's done.

I also have some experiments and ideas that I want to conduct using
extensions to this model and writing my own implementation is a good
starting point, I'll publish these should I get anywhere with them. I've
always written things from scratch (e.g. Wrote C++ contrastive divergence
for Restricted Boltzmann Machines years ago) so it's just in my nature.

Another motivation was to show what I can do. A lot of my work is either
stored away in backups or old disk images gathering dust and I often feel I
need to improve how I present it.

Most importantly though I wanted to catch the attention of @NVIDIA and have
them implement Weight Normalization and Layer Normalization in the CUDNN
implementations of LSTM and GRU. Would also be awesome if they even added
support for the more general form https://openreview.net/pdf?id=rk5upnsxe

A Tensorflow port is on my list but probably wouldn't outperform or be any
more elegant than the Theano implementation, I could be wrong but that's my
guess.
